graylog2-server/src/main/java/org/graylog2/migrations/V20180924111644_AddDefaultGrokPatterns.java

```diff
@@ -71,9 +72,13 @@ public void upgrade() {
                     .getResource("V20180924111644_AddDefaultGrokPatterns_Default_Grok_Patterns.json");
             final ContentPack contentPack = this.objectMapper.readValue(contentPackURL, ContentPack.class);
             final ContentPack pack = this.contentPackPersistenceService.insert(contentPack)
-                    .orElseThrow(() -> new Error("Content pack " + contentPack.id() + " with this revision " + contentPack.revision() + " already found!"));
+                    .orElseThrow(() -> new ContentPackException("Content pack " + contentPack.id() + " with this revision " + contentPack.revision() + " already found!"));
 
-            contentPackService.installContentPack(pack, Collections.emptyMap(), "Add default Grok patterns", "admin");
+            try {
+                contentPackService.installContentPack(pack, Collections.emptyMap(), "Add default Grok patterns", "admin");
+            } catch(ContentPackException e) {
+                LOG.warn("Could not install default grok patterns. The default grok patterns were already found and have been modified.");
```

Comment:
I think the second part of the error message is confusing:

> The default grok patterns were already found and have been modified.

To me that can read as: "someone modified the default grok patterns" or "we found the default grok patterns and modified them". Those two are different outcomes and I think we need to be clearer about it, even if we need to write a longer sentence.

As I mentioned in the original issue, we also don't say what the user should do once this happens. How about changing the text to something like:

"Could not install default grok patterns: the installation found some modified default grok patterns in your setup and did not update them. If you wish to use to the default grok patterns we provide, please delete the grok pattern <pattern name> and install the 'Default grok patterns' content pack manually."

As this migration will only run once, I would also make this an error so it has more visibility. If we don't know the pattern name that fails, it is also ok to show the whole stack trace.

Reply:
:+1: